extras/rya.geoindexing/src/main/java/org/apache/rya/indexing/geotemporal/model/EventQueryNode.java

```diff
@@ -0,0 +1,283 @@
+     */
+    private static void verifyAllPredicatesAreConstants(final Collection<StatementPattern> patterns) throws IllegalStateException {
+        requireNonNull(patterns);
+
+        for(final StatementPattern pattern : patterns) {
+            if(!pattern.getPredicateVar().isConstant()) {
+                throw new IllegalStateException("The Predicate of a Statement Pattern must be constant. Pattern: " + pattern);
+            }
+        }
+    }
+
+    @Override
+    public CloseableIteration<BindingSet, QueryEvaluationException> evaluate(final BindingSet bindings) throws QueryEvaluationException {
+        final List<BindingSet> list = new ArrayList<>();
+        try {
+            final Collection<Event> searchEvents;
+            final String subj;
+            // If the subject needs to be filled in, check if the subject variable is in the binding set.
+            if(subjectIsConstant) {
+                // if it is, fetch that value and then fetch the entity for the subject.
+                subj = subjectConstant.get();
+                searchEvents = eventStore.search(Optional.of(new RyaURI(subj)), Optional.of(geoFilters), Optional.of(temporalFilters));
+            } else {
+                searchEvents = eventStore.search(Optional.empty(), Optional.of(geoFilters), Optional.of(temporalFilters));
+            }
+
+            for(final Event event : searchEvents) {
+                final MapBindingSet resultSet = new MapBindingSet();
+                if(event.getGeometry().isPresent()) {
+                    final Geometry geo = event.getGeometry().get();
+                    final Value geoValue = ValueFactoryImpl.getInstance().createLiteral(geo.toText());
+                    final Var geoObj = geoPattern.getObjectVar();
+                    resultSet.addBinding(geoObj.getName(), geoValue);
+                }
+
+                final Value temporalValue;
+                if(event.isInstant() && event.getInstant().isPresent()) {
+                    temporalValue = ValueFactoryImpl.getInstance().createLiteral(event.getInstant().get().getAsDateTime().toString(TemporalInstantRfc3339.FORMATTER));
+                } else if(event.getInterval().isPresent()) {
+                    temporalValue = ValueFactoryImpl.getInstance().createLiteral(event.getInterval().get().getAsPair());
+                } else {
+                    temporalValue = null;
+                }
+
+                if(temporalValue != null) {
+                    final Var temporalObj = temporalPattern.getObjectVar();
+                    resultSet.addBinding(temporalObj.getName(), temporalValue);
+                }
+                list.add(resultSet);
+            }
+        } catch (final ObjectStorageException e) {
+            throw new QueryEvaluationException("Failed to evaluate the binding set", e);
+        }
+        if(bindings.size() != 0) {
+            list.add(bindings);
+        }
```

Comment:
You need to be careful here.  It is possible that your incoming BindingSet could contain a Binding that assigns a value to the subject variable of your StatementPattern constraints.  In this case, you need to include the subject as a constraint in your query.  The code in the evaluation method above needs to be placed inside an iteration that checks if the incoming BindingSet assigns a value to the subject variable.